src/org/intellij/grammar/generator/fleet/FleetParserGenerator.java

```diff
@@ -0,0 +1,178 @@
+    rootImport = getRootAttribute(psiFile, KnownAttribute.ELEMENT_TYPE_HOLDER_CLASS);
+    if (rootImport != null) myPossibleImports.add(rootImport);
+
+    myGenerateIFileType = generateIFileType;
+    myFileTypeClassName = classFileTypeName;
+    myFileTypeDebugName = debugFileTypeName;
+    myLanguageClass = languageClass;
+
+    myPossibleImports.add(myFileTypeClassName);
+    myPossibleImports.add(myLanguageClass);
+    myPossibleImports.add(myGrammarRootParser);
+  }
+
+  @Override
+  protected @NotNull GenOptions createGenerator(BnfFile file) {
+    return new GenOptions(file, true);
+  }
+
+  @Override
+  protected @NotNull String generatePackageName(String className) {
+    var packageName = super.generatePackageName(className);
+    if (G.adjustPackagesForFleet && !className.startsWith(FLEET_NAMESPACE)) {
+      if (packageName.isEmpty()) {
+        return FLEET_NAMESPACE;
+      }
+      return FLEET_NAMESPACE_PREFIX + packageName;
+    }
+
+    return packageName;
+  }
+
+  @Override
+  protected void generateParseMethod(String shortAN, String shortET, String root, String shortPB, String builder) {
+    out("public %s parse(%s %s, %s %s) {", shortAN, shortET, root, shortPB, builder);
+    out("throw new IllegalStateException(\"Use parseLight instead\");");
+    out("}");
+  }
+
+  @Override
+  protected @NotNull String adjustName(String original) {
+    switch (original) {
+      case BnfConstants.GPUB_CLASS:
+        return FleetConstants.GPUB_CLASS;
+      case BnfConstants.PSI_BUILDER_CLASS:
+        return FleetConstants.PSI_BUILDER_CLASS;
+      case BnfConstants.PSI_PARSER_CLASS:
+        return FleetConstants.PSI_PARSER_CLASS;
+      case BnfConstants.LIGHT_PSI_PARSER_CLASS:
+        return FleetConstants.LIGHT_PSI_PARSER_CLASS;
+      case BnfConstants.TOKEN_SET_CLASS:
+        return FleetConstants.TOKEN_SET_CLASS;
+      case BnfConstants.IELEMENTTYPE_CLASS:
+        return FleetConstants.IELEMENTTYPE_CLASS;
+      case BnfConstants.PSI_ELEMENT_CLASS:
+        return FleetConstants.PSI_ELEMENT_CLASS;
+      case BnfConstants.AST_NODE_CLASS:
+        return FleetConstants.AST_NODE_CLASS;
+      default:
+        break;
+    }
+
+    if (G.adjustPackagesForFleet &&
+        (original.equals(myVisitorClassName) || myPossibleImports.contains(original)) &&
+        !original.startsWith(FLEET_NAMESPACE)) {
+      return FLEET_NAMESPACE_PREFIX + original;
+    }
+
+    return original;
+  }
+
+  @Override
+  protected boolean useFactory(String factory) {
+    return false;
+  }
+
+  @Override
+  public @NotNull String shorten(@NotNull String s) {
+    return super.shorten(adjustName(s));
+  }
+
+  @Override
+  public void generate() throws IOException {
+    super.generate();
+    if (myGenerateIFileType) {
+      openOutput(adjustName(myFileTypeClassName));
+      try {
+        generateFileTypeClass();
+      }
+      finally {
+        closeOutput();
+      }
+    }
+  }
+
+  private void generateFileTypeClass() {
```

Comment:
1. Already outdated. `FleetPsiBuilder` is already gone.

2. Parser does not refer to this class, so it shall not be generated as part of parser generation. We can have a separate generator action for that if needed.